src/main/resources/avro/common.avdl

```diff
@@ -412,36 +365,21 @@ record Segment {
   out to the specified length.
   */
   long length;
-
-  /**
-  Start and end `Position`s where this `Segment` attaches to other sequences.
-  Note that the segmentId for start and end might not be the same. The
-  `Segment`s covering the sequences onto which this `Segment` is joined are
-  called its "parents", while this segment is a "child".
-
-  Joins may occur on the outer sides of the terminal bases in a sequence: the
-  left side of the base at index 0, and the right side of the base with maximum
-  index. These are the "terminal sides" of the sequence. `startJoin` is the join
-  on the side indicated by `start`, and may only be set if that side is a
-  terminal side. Similarly, `endJoin` is the join on the opposite side of the
-  piece of sequence selected by the segment, and may only be set if that side is
-  a terminal side. The value of `startJoin` or `endJoin`, if set, is the side to
-  which the corresponding side of this `Sequence` is connected.
-  */
-  union { null, Position } startJoin = null;
-  union { null, Position } endJoin = null;
 }
 
 /**
 A `Path` is an ordered list of `Segment`s. In general any contiguous path
 through a sequence graph, with no novel adjacencies, can be represented by a
 `Path`.
+
+Can optionally include the joins at either end of the path.
```

Comment:
It's an array of `Segment`s together with these optional end joins. It's to handle weird cases where you want to talk about a `Path` that includes only a `Join`, or a `Path` where it matters whether a join at one end is present or not.

This is sort of a fuzzy bit of the proposed changes; are there better ways to handle those cases, or good justifications for why we don't need them?